File: baselibrary/src/main/java/com/anningtex/baselibrary/base/AbsBaseFragment.java
package com.anningtex.baselibrary.base;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

/**
 * @ClassName: AbsBaseFragment
 * @Description: java类作用描述
 * @Author: alvis
 * @CreateDate: 2020/6/20 15:43
 */
public abstract class AbsBaseFragment<P extends BaseContract.Presenter> extends Fragment implements BaseContract.View {
    protected P mPresenter;
    protected Context mContext;
    protected WindowManager mWindowManager;
    protected DisplayMetrics mMetrics;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        this.mContext = context;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mWindowManager = (WindowManager) getContext().getSystemService(Context.WINDOW_SERVICE);
        mMetrics = new DisplayMetrics();
        mWindowManager.getDefaultDisplay().getMetrics(mMetrics);
        View view = inflater.inflate(getLayoutResId(), container, false);
        onFragmentViewCreated();
        return view;
    }

    public void onFragmentViewCreated() {

    }

    protected abstract int getLayoutResId();


    @Override
    public void toast(String msg) {
        Toast.makeText(mContext, msg, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void toast(int msg) {
        Toast.makeText(mContext, getString(msg), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        if (mPresenter != null) {
            mPresenter.detachView();
        }
    }

    public void $startActivity(Class<?> cls) {
        Intent intent = new Intent(getContext(), cls);
        startActivity(intent);
    }

    public void $startActivity(Class<?> cls, Bundle bundle) {
        Intent intent = new Intent(getContext(), cls);
        intent.putExtras(bundle);
        startActivity(intent);
    }

    public void $startActivity(Class<?> cls, String name, String value) {
        Intent intent = new Intent(getActivity(), cls);
        intent.putExtra(name, value);
        startActivity(intent);
    }

}

File: baselibrary/src/main/java/com/anningtex/baselibrary/base/AbsBaseFragment.kt
package com.anningtex.baselibrary.base

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.Toast
import androidx.fragment.app.Fragment

/**
 * @ClassName: AbsBaseFragment
 * @Description: java类作用描述
 * @Author: alvis
 * @CreateDate: 2020/6/20 15:43
 */
abstract class AbsBaseFragment<P : AbsBaseContract.Presenter<*>?> : Fragment(), AbsBaseContract.View {
    @JvmField
    protected var mPresenter: P? = null

    @JvmField
    protected var mContext: Context? = null

    @JvmField
    protected var mWindowManager: WindowManager? = null

    @JvmField
    protected var mMetrics: DisplayMetrics? = null
    override fun onAttach(context: Context) {
        super.onAttach(context)
        mContext = context
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mWindowManager = context!!.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        mMetrics = DisplayMetrics()
        mWindowManager!!.defaultDisplay.getMetrics(mMetrics)
        var view = inflater.inflate(layoutResId, container, false)
        onFragmentViewCreated(view)
        return view
    }

    protected abstract fun onFragmentViewCreated(view: View?)

    protected abstract val layoutResId: Int

    override fun toast(msg: String) {
        Toast.makeText(mContext, msg, Toast.LENGTH_SHORT).show()
    }

    override fun toast(msg: Int) {
        Toast.makeText(mContext, getString(msg), Toast.LENGTH_SHORT).show()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mPresenter?.detachView()
    }

    fun startAct(cls: Class<*>?) {
        val intent = Intent(context, cls)
        startActivity(intent)
    }

    fun startAct(cls: Class<*>?, bundle: Bundle?) {
        val intent = Intent(context, cls)
        intent.putExtras(bundle!!)
        startActivity(intent)
    }

    fun startAct(cls: Class<*>?, name: String?, value: String?) {
        val intent = Intent(activity, cls)
        intent.putExtra(name, value)
        startActivity(intent)
    }

    override fun showLoading() {}

    override fun dismissLoading() {}
}